pvlib/modelchain.py

```diff
@@ -349,6 +369,262 @@ def __init__(self, system, location,
         self.times = None
         self.solar_position = None
 
+    @classmethod
+    def with_pvwatts(cls, system, location,
+                     orientation_strategy=None,
+                     clearsky_model='ineichen',
+                     airmass_model='kastenyoung1989',
+                     temperature_model=None,
+                     name=None,
+                     **kwargs):
+        """
+        ModelChain that follows the PVWatts methods.
+
+        Parameters
+        ----------
+        system : PVSystem
+            A :py:class:`~pvlib.pvsystem.PVSystem` object that represents
+            the connected set of modules, inverters, etc.
+
+        location : Location
+            A :py:class:`~pvlib.location.Location` object that represents
+            the physical location at which to evaluate the model.
+
+        orientation_strategy : None or str, default None
+            The strategy for aligning the modules. If not None, sets the
+            ``surface_azimuth`` and ``surface_tilt`` properties of the
+            ``system``. Allowed strategies include 'flat',
+            'south_at_latitude_tilt'. Ignored for SingleAxisTracker systems.
+
+        clearsky_model : str, default 'ineichen'
+            Passed to location.get_clearsky.
+
+        airmass_model : str, default 'kastenyoung1989'
+            Passed to location.get_airmass.
+
+        temperature_model: None, str or function, default None
+            Valid strings are 'sapm', 'pvsyst', and 'faiman'. The ModelChain
+            instance will be passed as the first argument to a user-defined
+            function.
+
+        name: None or str, default None
+            Name of ModelChain instance.
+
+        **kwargs
+            Parameters supplied here are passed to the ModelChain
+            constructor and take precedence over the default
+            configuration.
+
+        Examples
+        --------
+        >>> module_parameters = dict(gamma_pdc=-0.003, pdc0=1)
+        >>> inverter_parameters = dict(pac0=1)
+        >>> system = PVSystem(surface_tilt=30, surface_azimuth=180,
+        ...     module_parameters=module_parameters,
+        ...     inverter_parameters=inverter_parameters)
+        >>> location = Location(32.2, -110.9)
+        >>> ModelChain.with_pvwatts(system, location)
+        ModelChain:
+          name: None
+          orientation_strategy: None
+          clearsky_model: ineichen
+          transposition_model: perez
+          solar_position_method: nrel_numpy
+          airmass_model: kastenyoung1989
+          dc_model: pvwatts_dc
+          ac_model: pvwatts_inverter
+          aoi_model: physical_aoi_loss
+          spectral_model: no_spectral_loss
+          temperature_model: sapm_temp
```

Comment:
How does `temperature_model=sapm_temp` come about? Its not the PVWatts default.

Related but different: the PVWatts temperature model is in a 1987 Sandia report. I could exhume it and add a pvlib function to support PVWatts as written, but, I think we could substitute the `sapm` temperature model for now, and use any of the predefined temperature_model_parameters.

Reply:
Added comment about this above `PVWATTS_CONFIG` definition. Copy/pasted the Fuentes reference from the PVWatts v5 documentation.

I agree that we should move forward assuming `sapm`. Should we make an issue to add the Fuentes model?

Reply:
Yes to adding the issue.

Reply:
#1032